application/src/test/java/com/sanction/thunder/resources/UserResourceTest.java

```diff
@@ -15,121 +15,116 @@
 
   @Test
-  public void testPostNullUser() {
-    Response response = resource.postUser(key, null);
-
-    assertEquals(Response.Status.BAD_REQUEST, response.getStatusInfo());
-  }
-
-  @Test
   public void testPostUserFailure() {
-    PilotUser user = mock(PilotUser.class);
-    when(usersDao.insert(user)).thenReturn(null);
+    PilotUser pilotUser = mock(PilotUser.class);
+    when(usersDao.insert(pilotUser)).thenReturn(null);
 
-    Response response = resource.postUser(key, user);
+    Response response = resource.postUser(pilotUser);
 
     assertEquals(Response.Status.CONFLICT, response.getStatusInfo());
   }
 
   @Test
   public void testPostUser() {
-    PilotUser user = mock(PilotUser.class);
-    when(usersDao.insert(user)).thenReturn(user);
+    PilotUser pilotUser = mock(PilotUser.class);
+    when(usersDao.insert(pilotUser)).thenReturn(pilotUser);
 
-    Response response = resource.postUser(key, user);
+    Response response = resource.postUser(pilotUser);
     PilotUser result = (PilotUser) response.getEntity();
 
     assertEquals(Response.Status.CREATED, response.getStatusInfo());
-    assertEquals(user, result);
+    assertEquals(pilotUser, result);
   }
 
   @Test
   public void testUpdateNullUser() {
-    Response response = resource.updateUser(key, null);
+    Response response = resource.updateUser(user, null);
+
+    assertEquals(Response.Status.BAD_REQUEST, response.getStatusInfo());
+  }
+
+  @Test
+  public void testUpdateUserMissmatch() {
+    User user = new User("wrong", "secret");
+    PilotUser pilotUser = new PilotUser("name", "secret", "", "", "");
+
+    Response response = resource.updateUser(user, pilotUser);
 
     assertEquals(Response.Status.BAD_REQUEST, response.getStatusInfo());
   }
 
   @Test
   public void testUpdateUserFailure() {
-    PilotUser user = mock(PilotUser.class);
-    when(usersDao.update(user)).thenReturn(null);
+    User user = new User("name", "secret");
+    PilotUser pilotUser = new PilotUser("name", "secret", "", "", "");
+    when(usersDao.update(pilotUser)).thenReturn(null);
 
-    Response response = resource.updateUser(key, user);
+    Response response = resource.updateUser(user, pilotUser);
 
     assertEquals(Response.Status.NOT_FOUND, response.getStatusInfo());
   }
 
   @Test
   public void testUpdateUser() {
-    PilotUser user = mock(PilotUser.class);
-    when(usersDao.update(user)).thenReturn(user);
+    User user = new User("name", "secret");
+    PilotUser pilotUser = new PilotUser("name", "secret", "", "", "");
+    when(usersDao.update(pilotUser)).thenReturn(pilotUser);
 
-    Response response = resource.updateUser(key, user);
+    Response response = resource.updateUser(user, pilotUser);
     PilotUser result = (PilotUser) response.getEntity();
 
     assertEquals(Response.Status.OK, response.getStatusInfo());
-    assertEquals(user, result);
-  }
-
-  @Test
-  public void testGetUserWithNullUsername() {
```

Comment:
Yea this one was because you can't have null credentials and that's where the username is pulled from